jupyterlab_bigquery/src/components/list_items_panel/service/bigquery_service.ts

```diff
@@ -63,4 +45,77 @@ export class BigQueryService {
         };
       });
   }
+
+  async listTables(projectId, datasetId): Promise<Partial<Dataset>> {
```

Comment:
You have the async modifier on the method but you use the older `.then(` based Promise handling. In general, it's best not to mix the two. I think async/await is more readable since you can use the more traditional try/catch block to handle any errors and you don't have to worry about callbacks after your asynchronous method resolves its data.

Reply:
Removed the `then` based Promise handling